### ISSUE TITLE
Allow choice value to be 0

### ISSUE THREAD

**Issue by NetanelBasal** [jshjohnson/Choices#309]
I have a choice which value is **0**, and it seems like nothing happens when I click this option. ( It does not trigger the `change` event )

```
const choices = [{
  value: 0,
  label: 'Label 1'
}]
```

I tried to change the value to any number other than zero, and it works fine.

**Comment by jeffperrin** [comment]
Looks like this is caused by this line: https://github.com/jshjohnson/Choices/blob/master/assets/scripts/src/choices.js#L1142

Anything 'falsey' will not trigger the change event (ie; 0 or an empty string). I'm not sure what the original reason for this check was, but it seems like it should explicitly be looking for null or undefined instead.

**Comment by NetanelBasal** [comment]
You are right. Thanks.

**Comment by Codecomp** [comment]
Just encountered this issue myself, had to mess around with several implementations to work around this issue.

**Comment by basgren** [comment]
It would be nice to allow any value to be used for choices. Even `null` or `undefined`. For example, in our application we have some parameters which can be set or unset, so we give user options like:
```js
[
    // Here we'd like to use `value: null` as this is consistent with our data model, 
    // but now we have to use -1 as a value and perform conversions `null` <--> -1 every 
    // time we need to sync data model and dropdown.
    {value: -1, label: '[Not used]'},
    {value: 1, label: 'Option 1'},
    {value: 2, label: 'Option 2'}
]
```

**Comment by basgren** [comment]
Looks like the issue with `0` value is fixed by #349, but it still restricts `null` or `undefined` values :)

**Comment by jonnyscholes** [comment]
Great to see this is fixed! Would it be possible to cherry pick this into master as a patch release? More than happy to submit a PR to master with just the changes from #349 if thats helpful :)
